### ISSUE TITLE
Page dimensions (width, height) are wrong for Page.getText("dict") (for "rawdict" as well) since version 1.18.6

### ISSUE THREAD

**Issue by pfille** [pymupdf/PyMuPDF#864]
## Describe the bug (mandatory)
After upgrading to PyMuPDF version 1.18.6 from 1.18.5 the page dimensions (width, height) as given in the dict and rawdict from Page.getText("dict") always equal 2.0

## To Reproduce (mandatory)
1. In new environment install PyMuPDF (I tested for Python 3.8.0 and 3.9.0, both on two different machines):
`$python -m pip install pymupdf`
Overview from '$pip list':
```
Package    Version
---------- -------
pip        20.3.3
PyMuPDF    1.18.6
setuptools 50.3.2
wheel      0.36.2

```
2. Get PDF file. I tested for [Microsoft's CSR report](https://query.prod.cms.rt.microsoft.com/cms/api/am/binary/RE4JaGo)
3. Get the page dimensions of any page from Page.getText("dict") (behavior is the same for "rawdict"):
```
import fitz

doc = fitz.open('Microsoft-2020-CSR-Report.pdf')
page = doc.loadPage(4)
dict_ = page.getText('dict')
print(f'w: {dict_["width"]}, h: {dict_["height"]}')

# Output:
# w: 2.0, h: 2.0
```
4. Downgrade to PymuPDF 1.18.5:
`$python -m pip uninstall pymupdf`
`$python - m pip install pymupdf==1.18.5`
overview from '$pip list':
```
Package    Version
---------- -------
pip        20.3.3
PyMuPDF    1.18.5
setuptools 50.3.2
wheel      0.36.2

```
5. Run same code as above:
```
import fitz

doc = fitz.open('Microsoft-2020-CSR-Report.pdf')
page = doc.loadPage(4)
dict_ = page.getText('dict')
print(f'w: {dict_["width"]}, h: {dict_["height"]}')

# Output:
# w: 1008.0, h: 612.0

```

## Expected behavior (optional)
It is expected that the values of the page dimensions stay the same for the most recent version. Also the value of 2.0 is not consistent with the coordinates of the bounding boxes of all items in the page (blocks, lines, etc.), which are typically much larger.

## Your configuration (mandatory)
 - Operating system: Ubuntu 18.04 and 20.04, both 64-bit
 - Python version 3.8.0 and 3.9.0, 64-bit
 - PyMuPDF version 1.18.6, installation method: wheel.

**Comment by JorjMcKie** [comment]
I had to fix a misbehaviour in MuPDF occurring in special cases. So I needed an indicator as to whether text extraction should occur unlimited (selection rect infinite) or confined to a certain rectangle.
The infinite rectangle is the one you are seeing if you do not specify any `clip`. If you do, then those values will again have meaningful values.
In addition, you do not really need those, right? They reflect the page rectangle.

**Comment by pfille** [comment]
Thank you very much for the quick response and your awesome work!
Now I can see, where this behavior is coming from and you are completely right that I can get the information from the page rectangle.
Do you agree that it might be a good idea to better reflect the current behavior in the documentation? For now, the wording in the description of `Page.getText()`, `TextPage`,  and the overview in "Dictionary Structure of `extractDICT()` and `extractRAWDICT()`" seem to imply that the width and height are valid information about the whole page, when omitting the `clip` option.

**Comment by JorjMcKie** [comment]
Thanks for your understanding.
You are right: I forgot to update the docu with this. But I also feel bad to carry around unusable stuff around. So I just implemented something which restores those two values - at least when text extraction is happening via the wrappers on page level. In absence of page information there is no way for the `TextPage` class to otherwise know the page dimension.
What is more: TextPage is not really connected to any document page and can deal with the amalgamated contents of several doc pages. Which is not used in PyMuPDF though.

So bottom line, in v1.18.7 the world is again in order - if nobody looks behind the curtain.
